multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -434,6 +499,8 @@ def infer_column_types(
             column_types[col_name] = CATEGORICAL
         elif is_numerical_column(data[col_name], valid_data[col_name]):  # Infer numerical column
             column_types[col_name] = NUMERICAL
+        elif is_document_column(data[col_name], col_name=col_name):
+            column_types[col_name] = DOCUMENT
```

Comment:
Consider making this check the last to avoid bringing overhead to other problem types, text or image classification.

Reply:
renamed is_document_column to is_document_image_column, and will add more inferences functions for other types of documents later.